apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -56,12 +135,57 @@ static void tc_compress_decompress(void)
 	TC_SUCCESS_RESULT();
 }
 
+static void tc_comp_decomp_buffer(void)
+{
+	int fd;
+	int ret_chk;
+	fd = get_compfd();
+
+	/*If FAIL : Failed to open testcase driver*/
+	TC_ASSERT_GEQ("get_compfd", fd, OK);
+	
+	// initializing struct to store compressed and decompressed data
+	ret_chk = initialize_tc_data();
+	TC_ASSERT_EQ("compression data initialize", ret_chk, OK);
+
+	// Test to check compression type 
+	ret_chk = ioctl(fd, COMPIOC_GET_COMP_TYPE, NULL);
+	TC_ASSERT_EQ_CLEANUP("compression_type_check", ret_chk, CONFIG_COMPRESSION_TYPE, uninitialize_tc_data());
+
+	// Test to check compression type string name
+	ret_chk = ioctl(fd, COMPIOC_GET_COMP_NAME, tc_comp_type_name);
+	TC_ASSERT_EQ_CLEANUP("compression type string check", ret_chk, OK, uninitialize_tc_data());
+	if ( CONFIG_COMPRESSION_TYPE == MINIZ_TYPE) {
+		TC_ASSERT_EQ_CLEANUP("compression type string check", strcmp(tc_comp_type_name, MINIZ_NAME), OK, uninitialize_tc_data());
+	} else {
+		TC_ASSERT_EQ_CLEANUP("compression type string check", strcmp(tc_comp_type_name, LZMA_NAME), OK, uninitialize_tc_data());
+	}	
+	
+	// Test compression method 
+	ret_chk = ioctl(fd, COMPIOC_COMPRESS, comp_tc_data);
+	TC_ASSERT_EQ_CLEANUP("compress block", ret_chk, OK, uninitialize_tc_data());
+	
+	// Test decompression method
+	ret_chk = ioctl(fd, COMPIOC_DECOMPRESS, decomp_tc_data);
+	TC_ASSERT_EQ_CLEANUP("decompress block", ret_chk, OK, uninitialize_tc_data());
+	
+	//check the data before compression and after decompression
+	for (int i = 0; i < comp_tc_data->input_size; i++){
+		TC_ASSERT_EQ_CLEANUP("output after decompress", comp_tc_data->input_buffer[i], decomp_tc_data->output_buffer[i], uninitialize_tc_data());
+	}
+
+	uninitialize_tc_data();
+	TC_SUCCESS_RESULT();
```

Comment:
We need to add test cases for positive as well as negative cases. 
Also, need to add test case for scenario where the output buffer is smaller than required size. In such case, we need to verify that complete data can be reconstructed after multiple calls to decompress api.